test_pull_requests

```diff
@@ -461,31 +413,107 @@ class FlushableStore
   end
 end
 
-# Quick-and-dirty middleware that attempts to invalidate appropriate
-# cache entries before handing the request off to the
-# Faraday::HttpCache middleware
-class GitHubCacheInvalidator < Faraday::Middleware
 
-  UNSAFE_METHODS = [:post, :put, :delete, :patch].freeze
+class InvalidatingGitHubCache < Faraday::HttpCache
+
+  class CustomInvalidatingStorage < Faraday::HttpCache::Storage
+
+    attr_accessor :invalidatable
+
+    def initialize(*args)
+      super
+      @invalidatable = Hash.new { |h,k| h[k] = {} }
+      @cache.storage = self
+    end
+
+    # GitHub-specific (although probably other APIs are similar) method
+    # to remove the pagination parameters from a URL so it can be better
+    # classified
+    #
+    # Example:
+    #   synthesize_url("https://api.github.com/some/resource/id?per_page=100")
+    #     returns
+    #   "https://api.github.com/some/resource/id"
+    def synthesize_url(orig_url)
```

Comment:
Doc and function name are misleading -- I'm a little confused but as I understand it we want to be doing two things:

 - storing `GET`s without their pagination queries
 - invalidating base `/comments` queries when a `DELETE` to a specific comment happens 
 - something similar to above for SHAs? Unclear

I think it would be better to have two methods

```rb
def non_paginated_url(original_url)

def more_generic_url(specific_url)
```

Those names suck but the point being it's two different transformations, right?